radix-engine/src/system/system_modules/costing/fee_reserve.rs

```diff
@@ -214,8 +214,8 @@ impl SystemLoanFeeReserve {
             effective_royalty_price,
 
             // System loan is used for both execution and royalty
-            xrd_balance: cost_unit_price * system_loan as u128,
-            xrd_owed: cost_unit_price * system_loan as u128,
+            xrd_balance: effective_execution_price * system_loan as u128,
+            xrd_owed: effective_execution_price * system_loan as u128,
```

Comment:
Use `effective_execution_price` to avoid situations where system loan is not enough for a `lock_fee` instruction due to high tip percentage.